app/registries/foreman/plugin.rb

```diff
@@ -466,12 +466,16 @@ def smart_proxies(klass)
       @smart_proxies.fetch(klass.name, {})
     end
 
-    def add_controller_action_scope(controller_class, action, &block)
-      controller_actions = @controller_action_scopes[controller_class.name] || {}
+    def add_controller_action_scope(controller_name, action, &block)
+      if controller_name.is_a? Class
```

Comment:
Maybe "not string" is better, but we probably going to drop this block after 1.22+ so no biggie.